sql/src/main/java/io/crate/execution/engine/collect/collectors/NodeStats.java

```diff
@@ -0,0 +1,188 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.collect.collectors;
+
+import io.crate.data.BatchIterator;
+import io.crate.data.CollectingBatchIterator;
+import io.crate.data.Row;
+import io.crate.exceptions.SQLExceptions;
+import io.crate.execution.dsl.phases.RoutedCollectPhase;
+import io.crate.execution.engine.collect.RowsTransformer;
+import io.crate.execution.engine.collect.stats.NodeStatsRequest;
+import io.crate.execution.engine.collect.stats.NodeStatsResponse;
+import io.crate.execution.engine.collect.stats.TransportNodeStatsAction;
+import io.crate.expression.InputFactory;
+import io.crate.expression.reference.StaticTableReferenceResolver;
+import io.crate.expression.reference.sys.node.NodeStatsContext;
+import io.crate.expression.symbol.RefVisitor;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.Reference;
+import io.crate.metadata.sys.SysNodesTableInfo;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.transport.ConnectTransportException;
+import org.elasticsearch.transport.ReceiveTimeoutTransportException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+
+/**
+ * Collects information for sys.nodes locally or remote based on the provided {@link RoutedCollectPhase}
+ */
+public final class NodeStats {
+
+    public static BatchIterator<Row> newInstance(TransportNodeStatsAction transportStatTablesAction,
+                                                 RoutedCollectPhase collectPhase,
+                                                 Collection<DiscoveryNode> nodes,
+                                                 InputFactory inputFactory) {
+
+        return CollectingBatchIterator.newInstance(
+            () -> {},
+            t -> {},
+            new LoadNodeStats(
+                transportStatTablesAction,
+                collectPhase,
+                nodes,
+                inputFactory
+            )
+        );
+    }
+
+    private static final class LoadNodeStats implements Supplier<CompletableFuture<? extends Iterable<? extends Row>>> {
+
+        private static final TimeValue REQUEST_TIMEOUT = TimeValue.timeValueMillis(3000L);
+        private final TransportNodeStatsAction transportStatTablesAction;
```

Comment:
I think the var name here is not valid anymore

Reply:
Changed it to `nodeStatsAction`